### ISSUE TITLE
default build options (strip, debug, optimization level)

### ISSUE THREAD

**Issue by rgommers** [FFY00/mesonpy#27]
Yet another default to choose here. My expectation is that the `meson.build` files go with Meson's default here (see https://mesonbuild.com/Builtin-options.html#core-options), because they're mostly used for local development.

For wheels however, it's a lot less likely that debugging is needed, and more likely the user wants small binary size and good performance. So perhaps the defaults in `mesonpy` should be:
- `-Ddebug=false`
- `-Dstrip=true`
- `-Doptimization=2` (note that `3` typically doesn't help much and may do some unsafe optimizations that lead to test failures)

Stripping binaries leads to ~3x size reduction, which for SciPy means we go from 400 MB to 100 MB installed size. Savings add up very quickly when doing that for multiple packages. We always strip wheels before uploading to PyPI.

**Comment by FFY00** [comment]
I agree with this. The only thing is that I think we should specify `-Dbuildtype=release` instead of `-Ddebug=false`, and allow the user to change it.

> `-Doptimization=2` (note that `3` typically doesn't help much and may do some unsafe optimizations that lead to test failures)

Yep, and does not necessarily result in faster code, as it basically means "enable all optimizations". Optimizations are a trade-off, and 2 is a curated list suitable for most people. 3 is useful in projects that benchmark and adjust their optimizations, like CPython with PGO. :laughing:

**Comment by rgommers** [comment]
> The only thing is that I think we should specify `-Dbuildtype=release`

I thought about that, it was my first idea actually. The one difference is that that defaults to `-O3`, and I thought `-O2` would be better. But I don't feel strongly about it either way.

**Comment by rgommers** [comment]
@FFY00 can we mark this for the next release? Should be an easy fix I'd think.

**Comment by FFY00** [comment]
It should be fixed in 0.4.0, I think we can close this now. I still want to allow users to change the default options, but that is a bit trickier, so I didn't do it right now.

**Comment by rgommers** [comment]
@FFY00 `0.4.0` was tagged but is missing from PyPI. Can you push it up?

**Comment by FFY00** [comment]
Oh, sorrt about that! :facepalm:

**Comment by rgommers** [comment]
No worries, thanks for the quick fix.
